src/games/strategy/engine/framework/startup/mc/GameSelectorModel.java

```diff
@@ -206,32 +195,28 @@ public void load(final File file, final Component ui) {
   }
 
   public boolean isSavedGame() {
-    return !m_fileName.endsWith(".xml");
+    return m_fileName == null || !m_fileName.endsWith(".xml");
   }
 
   private void error(final String message, final Component ui) {
     JOptionPane.showMessageDialog(JOptionPane.getFrameForComponent(ui), message, "Could not load Game", JOptionPane.ERROR_MESSAGE);
   }
 
-  public synchronized GameData getGameData() {
```

Comment:
@veqryn do you see or know of a reason for this class to actually be thread-safe? If so I can look that into the unit test I'm writing. From what I can tell the thread safetey requirements are mostly standard swing requirements, ie: watch out for changes on/from the dispatcher thread. Generally those threading issues are resolved by ensuring operations to shared objects are all done on one thread (removing synchronization needs, but you do then have to spawn up swing dispatcher thread events)